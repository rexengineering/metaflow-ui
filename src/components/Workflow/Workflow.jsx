import React from "react";
import PropTypes from "prop-types";
import { useSelector } from "react-redux";
import Task from "../Task";
import { selectTask } from "../../store/selectors";

function Workflow({ workflowID, className }) {
  const task = useSelector(selectTask(workflowID));
  return <Task className={className} task={task} />;
}

Workflow.propTypes = {
  className: PropTypes.string.isRequired,
  workflowID: PropTypes.string.isRequired,
};

export default Workflow;
