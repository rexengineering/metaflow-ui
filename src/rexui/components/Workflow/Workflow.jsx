import React, {useContext} from "react";
import { Button } from "@material-ui/core";
import { Context } from "../../Context";

function Workflow({ identifier, interactionId }){

 const {
     thunks: {
        cancelWorkflow
     }
 } = useContext(Context);

 const handleCancelWorkflow = () => cancelWorkflow(interactionId, identifier );

 return (
     <section>
       <p>Workflow: {identifier}</p>
      <Button size="small" onClick={handleCancelWorkflow} color="primary">Cancel</Button>
     </section>
 );

}

export default Workflow;
