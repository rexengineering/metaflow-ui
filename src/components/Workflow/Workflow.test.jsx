import { render, screen } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import Workflow from "./Workflow";
import getStore from "../../store";

describe("<Workflow />", () => {
  const renderWorkflow = (store = getStore({})) => (
    <Provider store={store}>
      <Workflow className="workflow" workflowID="456456" />
    </Provider>
  );

  it("should render", () => {
    const { asFragment } = render(renderWorkflow());
    const initialRender = asFragment();

    expect(initialRender).not.toBeNull();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(renderWorkflow());
    const initialRender = asFragment();

    expect(initialRender).toMatchSnapshot();
  });

  it("should render a title for the workflow", () => {
    render(renderWorkflow());
    const element = screen.getByText("456456");

    expect(element).toBeInTheDocument();
  });
});
