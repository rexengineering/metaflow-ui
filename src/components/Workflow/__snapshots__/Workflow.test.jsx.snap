// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Workflow /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 25px; height: 25px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</DocumentFragment>
`;
